core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -491,6 +491,21 @@ abstract class AbstractFetcherThread(name: String,
     } finally partitionMapLock.unlock()
   }
 
+  def maybeAddTopicIdsToThread(partitions: Set[TopicPartition], topicIds: String => Option[Uuid]) = {
+    partitionMapLock.lockInterruptibly()
+    try {
+      partitions.foreach { tp =>
+        val currentState = partitionStates.stateValue(tp)
+        if (currentState != null) {
+          val updatedState = currentState.updateTopicId(topicIds(tp.topic))
+          partitionStates.updateAndMoveToEnd(tp, updatedState)
```

Comment:
I think that we should use `update` here instead of `updateAndMoveToEnd`. `updateAndMoveToEnd` will basically deprioritize the partition and I don't think that we want to do this when we only update its topic id.

Reply:
Hmm. Update looks like it is private. The only method that calls it is "set" which clears the map and does various other things. I could try adding my own method though.

Reply:
Is this also updating and moving to the end? Will we have two versions in the map?

Reply:
What I mean is that updateAndMoveToEnd is the same except this new method does not remove the previous entry in the map.

Reply:
Right. `updateAndMoveToEnd` removes the item first and then adds it back at the end of the `LinkedHashMap`. We do rely on this to rotate the partitions in the `LinkedHashMap` in order to treat all partitions fairly. When we use a session, this is done on the broker as well btw.

The new `update` method will only update the state and will overwrite any existing state in the `LinkedHashMap`. This is what `map.put()` does. However, it does not change the order in the `LinkedHashMap` which is good. We don't have to change the oder when we set the topic id.

Reply:
I will do this. For some reason I thought put was like the queue/list data structure and not map. This makes sense to me now.